### ISSUE TITLE
Data correlation in k6

### ISSUE THREAD

**Issue by FloorD** [k6io/docs#328]
**How to correlate data in k6**

SrPerformo was concerned about data correlation in k6. He thinks old tools/UI tools are better are correlating data. (Session token, ViewState...).

He has an own methodology. 
- Recording the script twice with different users.
- Compare (diffing) the recorded session to find correlating tools.

https://www.srperf.com/scripting-03-spot-differences/

I think it is a nice idea to show one example about how to correlate data. 

SrPerformo believes other tools are better. I don't think k6 with (HAR recording) should be much more difficult than the other tools.

Show without the Chrome Inspector and with the Chrome Inspector

**Comment by imiric** [comment]
Good idea! I think there was an issue created by @dgzlopes that I can't find right now (maybe in the k6 repo?) that mentioned this. In theory this is possible right now by [exporting the end-of-test summary as JSON](https://k6.io/docs/results-visualization/end-of-test-summary#handlesummary-callback) and then comparing it for multiple runs using `jq` or some JSON diffing tool. But yeah, definitely should be documented.

Actually I think @dgzlopes created a Python script to do this, but I also can't find it, so he can chime in. :)

**Comment by dgzlopes** [comment]
Ned moved it to the `jslib` repo!

Here it's: https://github.com/k6io/jslib.k6.io/issues/28

The Python script was a quick one-off thing. I'm sure there is room for improvement :smile: 

btw, I've a ticket on Tempo related to this: https://github.com/grafana/tempo/issues/562

**Comment by na--** [comment]
@imiric and @dgzlopes, I am not sure you're talking about the same thing @FloorD is. Data correlation is not comparing the results of 2 test runs. It's figuring out what parts of the responses are dynamic (e.g. token, UUIDs, etc. in cookies and parameters) and have to be saved from previous responses and passed to future requests. 

And yeah, @FloorD, comparing 2 HAR recordings of the same scenario seems like a very nice way to figure these out! Though this issue might be better suited to https://github.com/k6io/har-to-k6, not the docs? `k6 convert` has a primitive `--correlate` flag, but we're trying to deprecate `k6 convert` altogether in favor of the newer `har-to-k6` tool.

**Comment by ppcano** [comment]
In docs, we have a [data correlation example / guide](https://k6.io/docs/examples/correlation-and-dynamic-data/). Whatever we improve in this area, it could also be added there.

**Comment by ppcano** [comment]
It would be nice to hear from @nicolevanderhoeven and @tom-miseur on this issue.

**Comment by markjmeier** [comment]
ping @tom-miseur, I know you have some thoughts on this!
